flumeserver/api/flume.go

```diff
@@ -1,166 +1,178 @@
 package api
 
 import (
-	// "sort"
-  // "bytes"
-  // "github.com/klauspost/compress/zlib"
-  // "strings"
-  // "time"
-  // "encoding/json"
-  // "math/big"
-  // "net/http"
-  "database/sql"
-  "github.com/ethereum/go-ethereum/common"
-  "github.com/ethereum/go-ethereum/common/hexutil"
-  // "github.com/ethereum/go-ethereum/core/types"
-  // "github.com/ethereum/go-ethereum/eth/filters"
-  // "github.com/ethereum/go-ethereum/rlp"
-  // "github.com/ethereum/go-ethereum/rpc"
-  // "io/ioutil"
-  // "io"
-  "context"
-  // "fmt"
-  "log"
-  // "sync"
+	"context"
+	"database/sql"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"log"
 )
 
 type FlumeAPI struct {
-	db *sql.DB
+	db      *sql.DB
 	network uint64
 }
 
-func NewFlumeAPI (db *sql.DB, network uint64 ) *FlumeAPI {
+func NewFlumeAPI(db *sql.DB, network uint64) *FlumeAPI {
 	return &FlumeAPI{
-		db: db,
+		db:      db,
 		network: network,
 	}
 }
 
 //offset should be a pointer to an int so that it is optional
 
-func (api *FlumeAPI) GetTransactionsBySender(ctx context.Context, address common.Address, offset int) (*paginator[map[string]interface{}], error) {
-
-  txs, err := getPendingTransactions(ctx, api.db, offset, 1000, api.network, "sender = ?", trimPrefix(address.Bytes()))
-  if err != nil {
-    log.Printf("Error getting pending txs: %v", err.Error())
-    return nil, err
-  }
-  ctxs, err := getTransactions(ctx, api.db, offset, 1000, api.network, "sender = ?", trimPrefix(address.Bytes()))
-  if err != nil {
-    log.Printf("Error getting txs: %v", err.Error())
-    return nil, err
-  }
-  txs = append(txs, ctxs...)
-  result := paginator[map[string]interface{}]{Items: txs}
-  if len(txs) >= 1000 {
-    result.Token = offset + len(txs)
-  }
-  return &result, nil
+func (api *FlumeAPI) GetTransactionsBySender(ctx context.Context, address common.Address, offset *int) (*paginator[map[string]interface{}], error) {
+
+	if offset == nil {
+		*offset = 0
```

Comment:
I'm pretty sure this will result in a nil pointer dereference. If `offset` is nil, you can't assign to `*offset`. Instead, you can do:

```
offset = new(int)
*offset = 0
```